### PR TITLE
Push and log search events to Google Analytics

### PR DESCRIPTION
**Pull request by ke4**
Log project searches on Google Analytics

### CODE REVIEW

**Comment by jacobwindsor** [comment]
Hi @sandykadam,
I don't think the VF component does what we require. We are sending custom events to google tag manager to track search terms and the results from those search terms. I can't see anything to allow us to do that with VF. Is that correct?

File: src/app/projects/projects-list/projects-list.component.ts [hunk 1]
```diff
@@ -174,5 +182,16 @@ export class ProjectsListComponent implements OnInit {
       ...this.filters.getValue(),
       searchVal: $search,
     });
+
+    const LIMIT = 5;
+
+    window['dataLayer'].push({
```
Comment (amnonkhen) [inline src/app/projects/projects-list/projects-list.component.ts:188]: I would put all the google analytics stuff in its own Angular Service.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Comment by sandykadam** [comment]
> Hi @sandykadam,
> I don't think the VF component does what we require. We are sending custom events to google tag manager to track search terms and the results from those search terms. I can't see anything to allow us to do that with VF. Is that correct?

Yes it does, I can show you quick demo of Covidportal where we tracking multiple events.

File: src/app/projects/projects-list/projects-list.component.ts (hunk 1 above)
Comment (ke4) [reply src/app/projects/projects-list/projects-list.component.ts:188]: Good suggestion. I was just waiting for the result of the conversation with Sandy.
After having the discussion around VF's GA component we decided to go with our own code as the VF component can't be used as we would like to.
So, I am going to do the service refactor now.

Comment (ke4) [reply src/app/projects/projects-list/projects-list.component.ts]: @amnonkhen  I extracted this code to a service and pushed it. Could you review it again please? Tx

**Review by amnonkhen — APPROVED** [review]

**Review by aaclan-ebi — APPROVED** [review]
LGTM!

**Pull request merged**